src/nvim/window.c

```diff
@@ -3541,6 +3541,19 @@ static void win_enter_ext(win_T *wp, bool undo_sync, int curwin_invalid, int tri
     }
     if (os_chdir((char *)curwin->w_localdir) == 0)
       shorten_fnames(TRUE);
+  
+  } else if (curtab->tp_localdir != NULL) {
+  	// Similar to above, but if the tab has a local directory.
+    if (globaldir == NULL) {
+      char_u cwd[MAXPATHL];
+
+      if (os_dirname(cwd, MAXPATHL) == OK) {
+        globaldir = vim_strsave(cwd);
+      }
+    }
+    if (os_chdir((char *)curtab->tp_localdir) == 0)
+      shorten_fnames(TRUE);
```

Comment:
`shorten_fnames` uses `int` as its parameter type, should I still pass a boolean anyway?